src/Slist.hs

```diff
@@ -944,6 +946,68 @@ break :: (a -> Bool) -> Slist a -> (Slist a, Slist a)
 break p = span (not . p)
 {-# INLINE break #-}
 
+{- | @O(n)@. Splits a 'Slist' into components of the given length. The last
+element may be shorter than the other chunks, depending on the length of the
+input.
+
+>>> chunksOf 3 $ slist [0..7]
+Slist {sList = [Slist {sList = [0,1,2], sSize = Size 3},Slist {sList = [3,4,5], sSize = Size 3},Slist {sList = [6,7], sSize = Size 2}], sSize = Size 3}
+>>> chunksOf 0 $ slist [0..10]
+Slist {sList = [], sSize = Size 0}
+>>> chunksOf (-13) $ slist [0..10]
+Slist {sList = [], sSize = Size 0}
+>>> chunksOf 100 $ slist [1,2,3]
+Slist {sList = [Slist {sList = [1,2,3], sSize = Size 3}], sSize = Size 1}
+
+>>> take 2 $ chunksOf 3 $ infiniteSlist [1..]
+Slist {sList = [Slist {sList = [1,2,3], sSize = Size 3},Slist {sList = [4,5,6], sSize = Size 3}], sSize = Size 2}
+
+@since x.x.x.x
+-}
+chunksOf :: Int -> Slist a -> Slist (Slist a)
+chunksOf i sl@Slist{..}
+    | i <= 0 = mempty
+    | sSize == Infinity = Slist (P.map (\x -> Slist x (Size i)) $ listChunksOf i sList) Infinity
```

Comment:
Maybe we can use `fromListN` from our `IsList` instance for a little bit of brevity 🙂 
```suggestion
    | sSize == Infinity = Slist (P.map (fromListN i) $ listChunksOf i sList) Infinity
```